Testsuite: gov.nasa.jpf.test.basic.HarnessTest
Tests run: 6, Failures: 0, Errors: 0, Time elapsed: 0.953 sec
------------- Standard Output ---------------
  running jpf with args:
JavaPathfinder core system v8.0 (rev 28+) - (C) 2005-2014 United States Government. All rights reserved.


====================================================== system under test
gov.nasa.jpf.test.basic.HarnessTest.runTestMethod()

====================================================== search started: 10/12/15 3:40 AM
** this is verifyAssert() - JPF should find an AssertionError

====================================================== error 1
gov.nasa.jpf.vm.NoUncaughtExceptionsProperty
java.lang.AssertionError: wrong answer..
	at gov.nasa.jpf.test.basic.HarnessTest.verifyAssert(HarnessTest.java:48)
	at java.lang.reflect.Method.invoke(gov.nasa.jpf.vm.JPF_java_lang_reflect_Method)
	at gov.nasa.jpf.util.test.TestJPF.runTestMethod(TestJPF.java:648)


====================================================== snapshot #1
thread java.lang.Thread:{id:0,name:main,status:RUNNING,priority:5,isDaemon:false,lockCount:0,suspendCount:0}
  call stack:
	at gov.nasa.jpf.util.test.TestJPF.runTestMethod(TestJPF.java:650)


====================================================== results
error #1: gov.nasa.jpf.vm.NoUncaughtExceptionsProperty "java.lang.AssertionError: wrong answer..  at gov.n..."

====================================================== search finished: 10/12/15 3:40 AM
  running jpf with args: +vm.class=InvalidVMClass +pass_exceptions
[SEVERE] JPF configuration error: class not found InvalidVMClass by classloader: gov.nasa.jpf.JPFClassLoader@5f8ed237
  running jpf with args:
JavaPathfinder core system v8.0 (rev 28+) - (C) 2005-2014 United States Government. All rights reserved.


====================================================== system under test
gov.nasa.jpf.test.basic.HarnessTest.runTestMethod()

====================================================== search started: 10/12/15 3:40 AM
** this is noViolation() - it should succeed

====================================================== results
no errors detected

====================================================== search finished: 10/12/15 3:40 AM
  running jpf with args:
JavaPathfinder core system v8.0 (rev 28+) - (C) 2005-2014 United States Government. All rights reserved.


====================================================== system under test
gov.nasa.jpf.test.basic.HarnessTest.runTestMethod()

====================================================== search started: 10/12/15 3:40 AM
** this is low level API test - it should succeed

====================================================== results
no errors detected

====================================================== search finished: 10/12/15 3:40 AM
  running jpf with args:
JavaPathfinder core system v8.0 (rev 28+) - (C) 2005-2014 United States Government. All rights reserved.


====================================================== system under test
gov.nasa.jpf.test.basic.HarnessTest.runTestMethod()

====================================================== search started: 10/12/15 3:40 AM
** this is verifyRuntimeException() - JPF should find an unhandled exception

====================================================== error 1
gov.nasa.jpf.vm.NoUncaughtExceptionsProperty
java.lang.RuntimeException: Bang!
	at gov.nasa.jpf.test.basic.HarnessTest.verifyRuntimeException(HarnessTest.java:68)
	at java.lang.reflect.Method.invoke(gov.nasa.jpf.vm.JPF_java_lang_reflect_Method)
	at gov.nasa.jpf.util.test.TestJPF.runTestMethod(TestJPF.java:648)


====================================================== snapshot #1
thread java.lang.Thread:{id:0,name:main,status:RUNNING,priority:5,isDaemon:false,lockCount:0,suspendCount:0}
  call stack:
	at gov.nasa.jpf.util.test.TestJPF.runTestMethod(TestJPF.java:650)


====================================================== results
error #1: gov.nasa.jpf.vm.NoUncaughtExceptionsProperty "java.lang.RuntimeException: Bang!  at gov.nasa.jpf..."

====================================================== search finished: 10/12/15 3:40 AM
  running jpf with args:
JavaPathfinder core system v8.0 (rev 28+) - (C) 2005-2014 United States Government. All rights reserved.


====================================================== system under test
gov.nasa.jpf.test.basic.HarnessTest.runTestMethod()

====================================================== search started: 10/12/15 3:40 AM
** this is verifyNullPointerException() - JPF should find an NPE

====================================================== error 1
gov.nasa.jpf.vm.NoUncaughtExceptionsProperty
java.lang.NullPointerException: Calling 'length()I' on null object
	at gov.nasa.jpf.test.basic.HarnessTest.verifyNullPointerException(HarnessTest.java:59)
	at java.lang.reflect.Method.invoke(gov.nasa.jpf.vm.JPF_java_lang_reflect_Method)
	at gov.nasa.jpf.util.test.TestJPF.runTestMethod(TestJPF.java:648)


====================================================== snapshot #1
thread java.lang.Thread:{id:0,name:main,status:RUNNING,priority:5,isDaemon:false,lockCount:0,suspendCount:0}
  call stack:
	at gov.nasa.jpf.util.test.TestJPF.runTestMethod(TestJPF.java:650)


====================================================== results
error #1: gov.nasa.jpf.vm.NoUncaughtExceptionsProperty "java.lang.NullPointerException: Calling 'length()I..."

====================================================== search finished: 10/12/15 3:40 AM
------------- ---------------- ---------------

Testcase: verifyAssert took 0.517 sec
Testcase: verifyJPFExcept took 0.005 sec
Testcase: noViolation took 0.126 sec
Testcase: testLowLevelAPI took 0.101 sec
Testcase: verifyRuntimeException took 0.094 sec
Testcase: verifyNullPointerException took 0.093 sec
